packages/eslint-plugin-bbva/docs/rules/prefer-throw-in-promise.md

```diff
@@ -0,0 +1,59 @@
+# Prefer throwing errors in promise handler functions (prefer-throw-in-promise)
+
+Prefer throwing `Error` instances instead of returning a new rejected promise to reject a promise chain.
+
+## Rule Details
+
+Examples of **incorrect** code for this rule:
+
+```js
+foo().then(() => {
+  return reject();
+});
+
+foo().then(..., () => {
+  return reject();
+});
+
+foo().catch(() => {
+  return reject();
+});
+
+try {
+  return reject();
+} catch (e) {
+  // ...
+}
+
+try {
+  // ...
+} catch (e) {
+  return reject();
```

Comment:
This should only be incorrect when it is inside an async function.

Reply:
Yep, I agree, thanks!